modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -83,115 +82,161 @@ def _read(cls, filepath_or_buffer, **kwargs):
                 skiprows=skiprows,
                 quotechar=quotechar,
                 is_quoting=is_quoting,
             )
-            for start, end in splits:
-                args.update({"start": start, "end": end})
-                partition_id = cls.deploy(cls.parse, num_splits + 2, args)
-                partition_ids.append(partition_id[:-2])
-                index_ids.append(partition_id[-2])
-                dtypes_ids.append(partition_id[-1])
-
-        # Compute the index based on a sum of the lengths of each partition (by default)
-        # or based on the column(s) that were requested.
+
+        partition_ids, index_ids, dtypes_ids = cls._launch_tasks(
+            splits, **partition_kwargs
+        )
+
+        new_query_compiler = cls._get_new_qc(
+            partition_ids=partition_ids,
+            index_ids=index_ids,
+            dtypes_ids=dtypes_ids,
+            index_col_md=index_col,
+            index_name=empty_pd_df.index.name,
+            column_widths=column_widths,
+            column_names=column_names,
+            squeeze=kwargs.get("squeeze", False),
+            skipfooter=kwargs.get("skipfooter", None),
+            parse_dates=kwargs.get("parse_dates", False),
+        )
+        return new_query_compiler
+
+    # _read helper functions
+    @classmethod
+    def _read_csv_check_support(
+        cls, filepath_or_buffer, read_csv_kwargs, compression_infered
+    ):
+        """
+        Check whatever or not passed parameters are supported by current modin.read_csv
+        implementation.
+        ----------
+        filepath_or_buffer:
+                `filepath_or_buffer` parameter of read_csv function.
+        read_csv_kwargs:
+                Parameters of read_csv function.
+        compression_infered:
+                Infered `compression` parameter of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whatever passed parameters are supported or not.
+        """
+        if isinstance(filepath_or_buffer, str):
+            if not cls.file_exists(filepath_or_buffer):
+                return False
+        elif not cls.pathlib_or_pypath(filepath_or_buffer):
+            return False
+
+        if compression_infered is not None:
+            use_modin_impl = compression_infered in ["gzip", "bz2", "xz"] or (
+                compression_infered == "zip"
+                # need python3.7 to .seek and .tell ZipExtFile
+                and sys.version_info[0] == 3
+                and sys.version_info[1] >= 7
+            )
+            if not use_modin_impl:
+                return False
+
+        if read_csv_kwargs.get("chunksize") is not None:
+            return False
+
+        skiprows = read_csv_kwargs.get("skiprows")
+        if skiprows is not None and not isinstance(skiprows, int):
+            return False
+
+        return True
+
+    @classmethod
+    def _define_index(cls, index_ids, index_col, index_name):
```

Comment:
Would it be possible to add type hints here?

Reply:
Yes, type hints are added.